### ISSUE TITLE
RNA insertion?

### ISSUE THREAD

**Issue by iaincambeul** [schmidt-lab/SPINE#5]
Hello,

I am wondering if there is any ability for this library to conduct domain insertion on RNA, given the nucleotide sequence to be inserted and the vector with the RNA that will be given the insertion?  If not, has anyone thought of any ways to modify the existing code to do this?

**Comment by nedru004** [comment]
Hello,

Thank you for your interest and for catching the error in our code. We have addressed the issue with default codon usage. As for creating a RNA domain insertion, that would not work with our system as we use double stranded sticky ends to create specific insertions.

David Nedrud

**Comment by iaincambeul** [comment]
Hello David,

Thanks for the quick response.  Maybe I should be clearer.  I am wondering if your insertion workflow would work on something a ribozyme encoded on a vector?  Is this something that double stranded sticky end ligation via Type II restriction enzymes would prevent?  

From first glance, it seems to me that as long as we are inserting between every nt instead of every 3 (between every codon), this should work on the DNA vector that encodes the ribozyme.

Best,
Ian

**Comment by nedru004** [comment]
Hello Ian,

Yes. I misunderstood. That would work with this workflow. We have designed this to insert between every codon and use the BsaI restriction enzyme. We can encode anything within the gene fragment, so that would be a simple edit to insert between every nucleotide. There is a loop that creates each oligo to be printed (OLS) and this loop iterates through each codon to insert the temporary genetic handle. This would need to be edited to loop through each nucleotide. Another note is that BsaI creates a 4 base overhang. We have discussed using SapI as it only leaves a 3 base overhang and it has been used in golden gate reactions in the past. This overhang is an unavoidable "linker" sequence that is fixed.

Let me know if you want help editing the code.

David
